content/en/docs/tasks/network/reconfigure-default-service-ip-ranges.md

```diff
@@ -0,0 +1,173 @@
+---
+reviewers:
+- thockin
+- dwinship
+min-kubernetes-server-version: v1.33
+title: Kubernetes Default Service CIDR Reconfiguration
+content_type: task
+---
+
+<!-- overview -->
+{{< feature-state feature_gate_name="MultiCIDRServiceAllocator" >}}
+
+This document shares how to reconfigure the default Service IP range(s) assigned
+to a cluster.
+
+## {{% heading "prerequisites" %}}
+
+{{< include "task-tutorial-prereqs.md" >}}
+
+{{< version-check >}}
+
+<!-- steps -->
+
+## Kubernetes Default Service CIDR Reconfiguration
+
+This document details how to reconfigure the existing Service IP address range
+within a Kubernetes cluster, which in turn affects the cluster's networking
+type, as outlined in the [Kubernetes Networking
+Concepts](https://kubernetes.io/docs/concepts/cluster-administration/
+networking/).
+
+The networking type of a Kubernetes cluster (IPv4 only, IPv6 only, or
+dual-stack) is largely determined by how the Services ip ranges are configured
+in the kube-apiserver (using the `--service-cluster-ip-range` flag). This
+configuration dictates the IP family (or families) from which ClusterIPs are
+allocated to Services. **The primary IP family configured in the kube-apiserver
+serves as the default IP family for allocating the Services ClusterIPs. It is
+important to note that Kubernetes can only allocate Service IPs from the IP
+families defined in the `--service-cluster-ip-range` flag.** The primary IP
+family of the cluster is typically represented by the IP family of the
+`kubernetes.default` Service ClusterIP.
+
+With the `MultiCIDRServiceAllocator` feature gate enabled, the default
+ServiceCIDR configuration is represented by a `ServiceCIDR` object named
+`kubernetes`. This object reflects the Service IP families configured for the
+cluster. The `kubernetes` `ServiceCIDR` object is created by the first
+kube-apiserver instance that starts after the feature gate is enabled, based on
+its configured `--service-cluster-ip-range` flag. Subsequent kube-apiserver
+instances will not attempt to update or delete this object, except for adding a
+new CIDR from a different IP family. This capability is essential for
+transitioning the cluster from a single-stack (IPv4-only or IPv6-only) to a
+dual-stack (IPv4/IPv6 or IPv6/IPv4) configuration
```

Comment:
```suggestion
kube-apiserver will create a `ServiceCIDR` object named `kubernetes` to
represent the CIDRs configured via `--service-cluster-ip-range`. This object will
remain unchanged after creation, unless you restart a formerly-single-stack
kube-apiserver with a dual-stack `--service-cluster-ip-range`, in which case it
will add the second CIDR to the `kubernetes` `ServiceCIDR`.
```

1. Don't mention the feature gate any more.
2. Don't talk about multiple apiservers, etc; that's not relevant to users.

Reply:
> Don't mention the feature gate any more.

yeah, absolutely

> Don't talk about multiple apiservers, etc; that's not relevant to users

his is the documentation that people that develop installers or deploy their own clusters use, shouldn't they be aware of these details?

Reply:
I guess we need to document it enough for that case, but it seemed like the original text was documenting too many minor implementation details...